src/Build/BuildCheck/Checks/PropertiesUsageCheck.cs

```diff
@@ -19,17 +19,17 @@ internal class PropertiesUsageCheck : InternalCheck
     private static readonly CheckRule _usedBeforeInitializedRule = new CheckRule("BC0201", "PropertyUsedBeforeDeclared",
         ResourceUtilities.GetResourceString("BuildCheck_BC0201_Title")!,
         ResourceUtilities.GetResourceString("BuildCheck_BC0201_MessageFmt")!,
-        new CheckConfiguration() { Severity = CheckResultSeverity.Warning, EvaluationCheckScope = EvaluationCheckScope.ProjectFileOnly });
+        new CheckConfiguration() { RuleId = "BC0201", Severity = CheckResultSeverity.Warning, EvaluationCheckScope = EvaluationCheckScope.ProjectFileOnly });
```

Comment:
we use the same id in 4 diff places per CheckRule definition.
Though it's not directly connected to your changes, but could you please move it to fields and reuse?